lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -24,47 +24,41 @@ define((require, exports, module) ->
     @UPDATE_DELAY: 1000
     @KATEX_OPTIONS = { displayMode: true, throwOnError: false }
 
-    @getMacrosArgumentRange: (session, argumentStartPos) ->
-      argumentRange = findSurroundingBrackets(session, argumentStartPos)
-      if argumentRange.mismatch
-        return null
-      else
-        # increment starting column to not include bracket
-        # do not decrement ending column because Range is left-open
-        return new Range(argumentRange.start.row, argumentRange.start.column + 1,
-                         argumentRange.end.row, argumentRange.end.column)
-
-    @getWholeEquation: (session, tokenIterator) ->
-      tokenValues = []
-      labelSequenceIndex = 0
-      labelParameters = []
-      curLabelParameter = null
-      curLabelTokens = []
-
-      range = tokenIterator.range
-      token = tokenIterator.getCurrentToken()
-      tokenPosition = tokenIterator.getCurrentTokenPosition()
-
-      while token?
-        acceptToken = true
-        if token.type == "storage.type.equation" and token.value == "\\label"
-          curLine = session.getLine(tokenPosition.row)
-          bracketPosition = tokenPosition.column + "\\label".length
-          if curLine[bracketPosition] == "{"
-            argumentRange = ContextHandler.getMacrosArgumentRange(session, { row: tokenPosition.row, column: bracketPosition + 1 })
-            if argumentRange?
-              acceptToken = false
-              labelParameters.push(session.getTextRange(argumentRange))
-              tokenIterator.stepTo(argumentRange.end.row, argumentRange.end.column + 1)
-
-        if acceptToken
-          tokenValues.push(token.value)
-
-        tokenIterator.stepForward()
-        token = tokenIterator.getCurrentToken()
-        tokenPosition = tokenIterator.getCurrentTokenPosition()
-
-      return { params: labelParameters, equation: tokenValues.join("") }
+    @extractEquation: (session, range) ->
+      { start, end } = range
+      joinedLines = (session.getLine(row) for row in [start.row..end.row]).join("\n")
+      startIndex = session.doc.positionToIndex(start, start.row)
+      endIndex = session.doc.positionToIndex(end, start.row) + 1
+      content = joinedLines.substring(startIndex, endIndex - 1)
+
+      labelRe = /\\label\s*\{/g
+      bracketString = "{"
+      labels = []
+      equationStrings = []
+      curIndex = 0
+
+      while true
+        result = labelRe.exec(content)
+        if result == null
+          equationStrings.push(content.substring(curIndex))
+          break
+
+        labelStartIndex = result.index
+        equationStrings.push(content.substring(curIndex, labelStartIndex))
+
+        matchedString = result[0]
+        openingBracketIndex = labelStartIndex + matchedString.length - 1
```

Comment:
Thanks, I've found another bug :)
When we determine the equation range and stumble upon a comment, we stop reading further. I should fix it.